codex-rs/tui/src/bottom_pane/mod.rs

```diff
@@ -281,15 +335,84 @@ impl BottomPane<'_> {
         self.composer.on_file_search_result(query, matches);
         self.request_redraw();
     }
+
+    /// Set the rows and cap for the transient live ring overlay.
+    pub(crate) fn set_live_ring_rows(
+        &mut self,
+        max_rows: u16,
+        rows: Vec<ratatui::text::Line<'static>>,
```

Comment:
`use ratatui::text::Line` up top?

```suggestion
        rows: Vec<Line<'static>>,
```